archaius-core/src/main/java/com/netflix/config/ConcurrentCompositeConfiguration.java

```diff
@@ -109,15 +110,19 @@ public class ConcurrentCompositeConfiguration extends ConcurrentMapConfiguration
     private final boolean enableStackTrace = Boolean.parseBoolean(System.getProperty(ENABLE_STACK_TRACE));
     private final boolean enableInstrumentation = Boolean.parseBoolean(System.getProperty(ENABLE_INSTRUMENTATION));
 
-    private Map<String, AbstractConfiguration> namedConfigurations = new ConcurrentHashMap<String, AbstractConfiguration>();
+    private Map<String, AbstractConfiguration> namedConfigurations = new ConcurrentHashMap<>();
 
     private final Map<String, Integer> stackTraces = new ConcurrentHashMap<>();
-    private final Set<String> usedProperties = ConcurrentHashMap.newKeySet();
+    private final AtomicReference<Set<String>> usedPropertiesRef = new AtomicReference<>(ConcurrentHashMap.newKeySet());
 
     public Set<String> getUsedProperties() {
-        return usedProperties;
+        return Collections.unmodifiableSet(new HashSet<>(usedPropertiesRef.get()));
     }
 
+    public Set<String> getAndClearUsedProperties() {
+        Set<String> ret = usedPropertiesRef.getAndSet(ConcurrentHashMap.newKeySet());
+        return Collections.unmodifiableSet(new HashSet<>(ret));
```

Comment:
Same comment about double wrapping as in the other PR.

Reply:
Removed the double wrap here to just directly return the unmodifiableSet